buildSrc/src/main/kotlin/io/spine/internal/gradle/git/Repository.kt

```diff
@@ -159,17 +159,14 @@ class Repository : AutoCloseable {
          *
          * @throws IllegalArgumentException if SSH URL is an empty string.
          */
-        fun of(sshUrl: String, user: Config.User, branch: String = Branch.master): Repository {
-            Preconditions.checkArgument(
-                !sshUrl.isBlank(),
-                "SSH URL cannot be an empty string."
-            )
+        fun of(sshUrl: String, user: UserInfo, branch: String = Branch.master): Repository {
```

Comment:
@tripolkaandrey, if it comes to such a static factory method (or even several similar ones), having a named factory object in Kotlin is too much of a ceremony. 

In Java we would write `UserInfo.of(...)` using a static factory method, and it feels natural. Why have `UserInfo.Factory.of(...)` in Kotlin? It's longer to read and longer to type.

A named `Factory` class (rather than object) would be appropriate when we have some parameters passed to it. Or, we have the state of the factory, so it would start producing slightly different objects.

Otherwise, static methods are just fine.

Reply:
I used `.Factory.` explicitly because it avoids hiding the type's documentation. Otherwise, when you have `UserInfo.of(...)` and you hover over the `UserInfo` part there is documentation about the `UserInfo.Factory` instead of the `UserInfo`.